hazelcast/src/main/java/com/hazelcast/client/impl/proxy/NearCachedClientMapProxy.java

```diff
@@ -481,6 +482,26 @@ public LocalMapStats getLocalMapStats() {
         return localMapStats;
     }
 
+    /**
+     * Async version of {@link #executeOnKeys}.
+     */
+    public <R> InternalCompletableFuture<Map<K, R>> submitToKeys(@Nonnull Set<K> keys,
+                                                                 @Nonnull EntryProcessor<K, V, R> entryProcessor) {
+        checkNotNull(keys, NULL_KEY_IS_NOT_ALLOWED);
+
+        Set<?> nearCacheKeys = serializeKeys
+            ? keys.stream().map(this::toNearCacheKey).collect(Collectors.toSet())
+            : keys;
+
+        InternalCompletableFuture<Map<K, R>> response;
+        try {
+            response = super.submitToKeys(keys, entryProcessor);
```

Comment:
if `serializeKeys` is true, we serialize all keys to convert them to near-cache keys. But here, in `super.submitToKeys`, we also do an extra serialization for all passed `keys` before sending them to server. This means we do same thing twice. This could have been done as in this commit: https://github.com/hazelcast/hazelcast/commit/57f09edf8015099eeefb1a98eb9efbbfaf30f060

Can you please also address this double serialization problem?

Reply:
patch is applied. Thanks